src/App.tsx

```diff
@@ -0,0 +1,155 @@
+    e.preventDefault();
+    if (message) {
+      setActive(true);
+      window.setTimeout(hideToast, timer);
+    }
+  };
+
+  return (
+    <main className={styles.main}>
+      <header className={styles.hero}>
+        <h1 className={styles.heading}>React flash notifications</h1>
+        <p className={styles.paragraph}>
+          A simple React component to display flash messages
+        </p>
+        <a
+          href="https://github.com/claeusdev/react-flash-message"
+          target="_blank"
+          rel="noopener noreferrer"
+          className={styles.link}
+        >
+          View on Github
+        </a>
+      </header>
+      <form onSubmit={handleShowToast} className={styles.testForm}>
+        <div className={styles.options}>
+          <div className={styles.dropdownGroup}>
+            <label htmlFor="type" className={styles.dropdownLabel}>
+              Type:
+            </label>
+            <select
+              id="type"
+              onChange={({ target }) =>
+                setType(target.value as unknown as Type)
+              }
+              value={type}
+              className={styles.dropdown}
+            >
+              <option value="default">Default</option>
+              <option value="warning">Warning</option>
+              <option value="success">Success</option>
+              <option value="error">Error</option>
+            </select>
+          </div>
+          <div className={styles.dropdownGroup}>
+            <label htmlFor="width" className={styles.dropdownLabel}>
+              Width:
+            </label>
+            <select
+              id="width"
+              onChange={({ target }) =>
+                setWidth(target.value as unknown as Width)
+              }
+              value={width}
+              className={styles.dropdown}
+            >
+              <option value="default">Default</option>
+              <option value="small">Small</option>
+              <option value="medium">Medium</option>
+              <option value="large">Large</option>
+              <option value="full">Full</option>
+            </select>
+          </div>
+          <div className={styles.dropdownGroup}>
+            <label htmlFor="position" className={styles.dropdownLabel}>
+              Position:
+            </label>
+            <select
+              id="position"
+              onChange={({ target }) =>
+                setPosition(target.value as unknown as Position)
+              }
+              value={position}
+              className={styles.dropdown}
+            >
+              <option value="default">Default</option>
+              <option value="tleft">Top Left</option>
+              <option value="tright">Top Right</option>
+              <option value="tcenter">Top Center</option>
+              <option value="bleft">Bottom Left</option>
+              <option value="bright">Bottom Right</option>
+              <option value="bcenter">Bottom Center</option>
```

Comment:
we can make this seven simpler by putting these values in an `enum` or list and then looping through to display.

```ts
const SELECT_VALUES = [
  {
    title: "Default",
    value: "left"
  }
]
```

Then, we can do: 
`SELECT_VALUES.map(value => <option value={value}>{title}</option>)`

Reply:
import React from "react";
import styles from ".././App.module.css";
import { Position, PositionSelectorProps } from "../utils/types";

const PositionSelector = ({ setPosition, position }: PositionSelectorProps) => {
  const positions = [
    { value: "default", title: "Default" },
    { value: "tleft", title: "Top Left" },
    { value: "tright", title: "Top Right" },
    { value: "tcenter", title: "Top Center" },
    { value: "bleft", title: "Bottom left" },
    { value: "bright", title: "Bottom Right" },
    { value: "bcenter", title: "Bottom Center" },
  ];

  return (
    <>
      {" "}
      <div className={styles.dropdownGroup}>
        <label htmlFor="position" className={styles.dropdownLabel}>
          Position:
        </label>
        <select
          id="position"
          onChange={({ target }) =>
            setPosition(target.value as unknown as Position)
          }
          value={position}
          className={styles.dropdown}
        >
          {positions.map(
            ({ value, title }: { value: string; title: string }) => (
              <option value={value}>{title}</option>
            )
          )}
        </select>
      </div>{" "}
    </>
  );
};

export default PositionSelector;

Reply:
Very hard to read te code. could you format it?